vertex/vertex.go

```diff
@@ -89,15 +97,21 @@ func vertexMiddleware(region, projectID string) sdkoption.Middleware {
 					specifier = "streamRawPredict"
 				}
 
+				if strings.Contains(model, "/") {
+					model = strings.SplitN(model, "/", 2)[1]
+				}
+
 				r.URL.Path = fmt.Sprintf("/v1/projects/%s/locations/%s/publishers/anthropic/models/%s:%s", projectID, region, model, specifier)
-			}
 
-			if r.URL.Path == "/v1/messages/count_tokens" && r.Method == http.MethodPost {
+			case r.URL.Path == "/v1/messages/count_tokens" && r.Method == http.MethodPost:
 				if projectID == "" {
 					return nil, fmt.Errorf("no projectId was given and it could not be resolved from credentials")
 				}
 
 				r.URL.Path = fmt.Sprintf("/v1/projects/%s/locations/%s/publishers/anthropic/models/count-tokens:rawPredict", projectID, region)
+
+			default:
+				return nil, fmt.Errorf("vertex middleware does not support %s %s", r.Method, r.URL.Path)
```

Comment:
Looks like if we miss any path, we probably forgot something in the code. A good idea to error here?